digitaltwins/samples/device/javascript/pnpSampleDevice.js

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+const Protocol = require('azure-iot-device-mqtt').Mqtt;
+const Client = require('azure-iot-device').Client;
+const Message = require('azure-iot-device').Message;
+
+const packageJson = require('./package.json');
+
+// String containing Hostname, Device Id & Device Key in the following formats:
+//  'HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>'
+const deviceConnectionString = process.env.DEVICE_CONNECTION_STRING;
+
+const modelId = 'dtmi:com:example:SampleDevice;1';
+const messageSubjectProperty = '$.sub';
+const sensorComponentName = 'sensor';
+const deviceInfoComponentName = 'deviceInformation';
+const sdkInfoComponentName = 'sdkInformation';
+const componentPrefix = '$iotin:';
+const commandComponentCommandNameSeparator = '*';
+
+const commandNameBlink = componentPrefix + sensorComponentName + commandComponentCommandNameSeparator + 'blink';
+const commandNameTurnOn = componentPrefix + sensorComponentName + commandComponentCommandNameSeparator + 'turnOn';
+const commandNameTurnOff = componentPrefix + sensorComponentName + commandComponentCommandNameSeparator + 'turnOff';
+
+const propertyUpdateHandler = (deviceTwin, componentName, propertyName, reportedValue, desiredValue, version) => {
+  console.log('Received an update for component: ' + componentName + ' and property: ' + propertyName + ' with value: ' + JSON.stringify(desiredValue));
+  const patch = helperCreateReportedPropertiesPatch(
+    { [propertyName]: desiredValue },
+    componentName,
+    {
+      code: 200,
+      description: 'Successfully executed patch for ' + propertyName,
+      version: version
+    }
+  );
+  updateComponentReportedProperties(deviceTwin, patch, componentName);
+  console.log('updated the property');
+};
+
+
+const commandHandler = async (request, response) => {
+  switch (request.methodName) {
+  case commandNameBlink: {
+    await sendCommandResponse(request, response, 200, 'blink response');
+    break;
+  }
+  case commandNameTurnOn: {
+    await sendCommandResponse(request, response, 200, 'turn On response');
+    break;
+  }
+  case commandNameTurnOff: {
+    await sendCommandResponse(request, response, 200, 'turn Off response');
+    break;
+  }
+  default: // Is there no way to register for an unknown method?
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+// TODO Not sure if this needs to be async or not coz we probably dont care if the response rached Hub.
+const sendCommandResponse = async (request, response, status, payload) => {
+  helperLogCommandRequest(request);
+  try {
+    await response.send(status, payload);
```

Comment:
>response.send [](start = 10, length = 13)

This is written to be support callbacks or promises so it is perfectly reasonable for you to use await in this async function.